tests/text/test_tabstops.py

```diff
@@ -34,9 +34,30 @@ def it_can_iterate_over_its_tab_stops(self, iter_fixture):
         assert count == expected_count
         assert TabStop_.call_args_list == expected_calls
 
+    def it_can_get_a_tab_stop_by_index(self, index_fixture):
+        tab_stops, expected_vals, tab_stop_, TabStop_, expected_calls = (
+            index_fixture
+        )
+        for count in range(len(tab_stops)):
+            assert tab_stops[count] is tab_stop_
+        assert TabStop_.call_args_list == expected_calls
+
```

Comment:
1. You're going to need two tests here, one for the tabs is non-empty case and one for the empty case (where it raises an exception).

2. I expect this test came from somewhere else in the code base which means I probably wrote its progenitor, but I'm inclined toward a straight-through approach here where one test case calls `tab_stops[idx]` once and only once. That makes the test method as simple and straightforward as possible. If there is merit in it being called more than once on the same XML structure, that can be implemented as multiple cases (params) in the fixture.

  ```python
  def it_can_get_a_tab_stop_by_index(self, index_fixture):
      tab_stops, idx, TabStop_, tab, tab_stop_ = index_fixture

      tab_stop = tab_stops[idx]

      TabStop_.assert_called_once_with(tab)
      assert tab_stop is tab_stop_
  ```

3. For what it may be worth, there's a non-sequential order in which I write these tests. I start with the test, then fixture comes second. For the test, I start with the action, calling `tab_stops[idx]` in this case. Next I move on to what I want to assert about that. Then I look for variables I need (flake8 built into my editor (vim) highlights those for me), and I have them come from the fixture, in the order they're needed. So middle, bottom, top order, in which the first step is easy and each step leads to the next.

  On the fixture, I start with the return statement, exactly matching the LHS of the assignment in the test. Then I work out how to get those items one by one. It's remarkably mechanical once you get the hang of it; the same patterns over and over.